### ISSUE TITLE
gpg agent failing with "Odd-length string"

### ISSUE THREAD

**Issue by lukechilds** [romanz/trezor-agent#363]
When running `gpg` I get a `Broken pipe` error:

```
$ echo hi | gpg --sign
gpg: using "Luke Childs <lukechilds123@gmail.com>" as default secret key for signing
gpg: problem with fast path key listing: End of file - ignored
gpg: keydb_search failed: Broken pipe
gpg: no default secret key: Broken pipe
gpg: signing failed: Broken pipe
```

Logs from `trezor-gpg-agent`:

```
2021-10-14 18:58:00,939 DEBUG        accepted connection on /Users/lukechilds/.gnupg/trezor/S.gpg-agent                                   [agent.py:20]
2021-10-14 18:58:00,939 DEBUG        <- b'OK'                                                                                             [keyring.py:53]
2021-10-14 18:58:00,939 DEBUG        -> b'RESET'                                                                                          [keyring.py:71]
2021-10-14 18:58:00,939 DEBUG        <- b'OK'                                                                                             [keyring.py:53]
2021-10-14 18:58:00,939 DEBUG        -> b'OPTION ttytype=screen-256color'                                                                 [keyring.py:71]
2021-10-14 18:58:00,939 DEBUG        options: [b'ttytype=screen-256color', b'allow-pinentry-notify', b'agent-awareness=2.1.0', b'ttytype=screen-256color', b'allow-pinentry-notify', b'agent-awareness=2.1.0', b'ttytype=screen-256color'] [agent.py:117]
2021-10-14 18:58:00,939 DEBUG        <- b'OK'                                                                                             [keyring.py:53]
2021-10-14 18:58:00,946 DEBUG        -> b'GETINFO version'                                                                                [keyring.py:71]
2021-10-14 18:58:00,946 DEBUG        <- b'D 2.3.2'                                                                                        [keyring.py:53]
2021-10-14 18:58:00,946 DEBUG        <- b'OK'                                                                                             [keyring.py:53]
2021-10-14 18:58:00,946 DEBUG        -> b'OPTION allow-pinentry-notify'                                                                   [keyring.py:71]
2021-10-14 18:58:00,946 DEBUG        options: [b'ttytype=screen-256color', b'allow-pinentry-notify', b'agent-awareness=2.1.0', b'ttytype=screen-256color', b'allow-pinentry-notify', b'agent-awareness=2.1.0', b'ttytype=screen-256color', b'allow-pinentry-notify'] [agent.py:117]
2021-10-14 18:58:00,953 DEBUG        <- b'OK'                                                                                             [keyring.py:53]
2021-10-14 18:58:00,954 DEBUG        -> b'OPTION agent-awareness=2.1.0'                                                                   [keyring.py:71]
2021-10-14 18:58:00,954 DEBUG        options: [b'ttytype=screen-256color', b'allow-pinentry-notify', b'agent-awareness=2.1.0', b'ttytype=screen-256color', b'allow-pinentry-notify', b'agent-awareness=2.1.0', b'ttytype=screen-256color', b'allow-pinentry-notify', b'agent-awareness=2.1.0'] [agent.py:117]
2021-10-14 18:58:00,954 DEBUG        <- b'OK'                                                                                             [keyring.py:53]
2021-10-14 18:58:00,954 DEBUG        -> b'SCD SERIALNO'                                                                                   [keyring.py:71]
2021-10-14 18:58:00,958 DEBUG        <- b'ERR 100696144 No such device <SCD>'                                                             [keyring.py:53]
2021-10-14 18:58:00,959 DEBUG        -> b'KEYINFO 680902703571767C4751893D772E1EB049BD0692'                                               [keyring.py:71]
2021-10-14 18:58:00,959 DEBUG        <- b"S KEYINFO b'680902703571767C4751893D772E1EB049BD0692' X - - - - - - -"                          [keyring.py:53]
2021-10-14 18:58:00,959 DEBUG        <- b'OK'                                                                                             [keyring.py:53]
2021-10-14 18:58:00,960 DEBUG        -> b'HAVEKEY --list=1000'                                                                            [keyring.py:71]
2021-10-14 18:58:00,960 ERROR        handler failed: Odd-length string                                                                    [__init__.py:271]
Traceback (most recent call last):
  File "/opt/homebrew/Cellar/trezor-agent/0.11.0_2/libexec/lib/python3.9/site-packages/libagent/gpg/__init__.py", line 263, in run_agent
    handler.handle(conn)
  File "/opt/homebrew/Cellar/trezor-agent/0.11.0_2/libexec/lib/python3.9/site-packages/libagent/gpg/agent.py", line 242, in handle
    handler(conn, args)
  File "/opt/homebrew/Cellar/trezor-agent/0.11.0_2/libexec/lib/python3.9/site-packages/libagent/gpg/agent.py", line 102, in <lambda>
    b'HAVEKEY': lambda _, args: self.have_key(*args),
  File "/opt/homebrew/Cellar/trezor-agent/0.11.0_2/libexec/lib/python3.9/site-packages/libagent/gpg/agent.py", line 205, in have_key
    self.get_identity(keygrip=keygrip)
  File "/opt/homebrew/Cellar/trezor-agent/0.11.0_2/libexec/lib/python3.9/site-packages/libagent/util.py", line 230, in wrapper
    result = method(self, *args, **kwargs)
  File "/opt/homebrew/Cellar/trezor-agent/0.11.0_2/libexec/lib/python3.9/site-packages/libagent/gpg/agent.py", line 161, in get_identity
    keygrip_bytes = binascii.unhexlify(keygrip)
binascii.Error: Odd-length string
```

You can see it's failing here with `Odd-length string`.

Any ideas?

**Comment by romanz** [comment]
Many thanks for reporting this issue!
Which `gpg` version are you running?

**Comment by lukechilds** [comment]
Thanks for taking a look!

>Which gpg version are you running?

```
$ gpg --version
gpg (GnuPG) 2.3.2
libgcrypt 1.9.4
Copyright (C) 2021 Free Software Foundation, Inc.
License GNU GPL-3.0-or-later <https://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Home: /Users/lukechilds/.gnupg
Supported algorithms:
Pubkey: RSA, ELG, DSA, ECDH, ECDSA, EDDSA
Cipher: IDEA, 3DES, CAST5, BLOWFISH, AES, AES192, AES256, TWOFISH,
        CAMELLIA128, CAMELLIA192, CAMELLIA256
AEAD: EAX, OCB
Hash: SHA1, RIPEMD160, SHA256, SHA384, SHA512, SHA224
Compression: Uncompressed, ZIP, ZLIB, BZIP2
```

**Comment by romanz** [comment]
Fixed in #365
